sched/misc/deadlock.c

```diff
@@ -100,8 +100,8 @@ static void collect_deadlock(FAR struct tcb_s *tcb, FAR void *arg)
       pid_t holder;
       size_t i;
 
-      holder = mutex->holder;
-      if (holder == NXMUTEX_NO_HOLDER)
+      holder = nxmutex_get_holder(mutex);
+      if (holder < 0)
```

Comment:
Looks like legacy but I don't quite understand this special -1 value. Why cant 0 mean no holder ?

Reply:
-1 is used in many places, like in pthread_mutex implementation, where there is comparison if (holder < 0). Why change these all, as pid_t is signed and the special value can be the same as before?